### PR TITLE
Updated WPZaakService

### PR DESCRIPTION
**Pull request by smisidjan**

### CODE REVIEW

**Review by rjzondervan — APPROVED** [review]

File: Installation/Action/waar.WaardepapierenDecosZaakAction.action.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "title": "WaardepapierenDecosZaakAction",
+  "$id": "https://waardepapieren.commonground.nl/action/waar.WaardepapierenDecosZaakAction.action.json",
+  "$schema": "https://docs.commongateway.nl/schemas/Action.schema.json",
+  "version": "0.0.10",
```
Comment (bbrands02) [inline Installation/Action/waar.WaardepapierenDecosZaakAction.action.json]: if its new the version could be 0.0.1?

File: Installation/Action/waar.WaardepapierenZaakAction.action.json [hunk 2]
```diff
@@ -2,39 +2,30 @@
   "title": "WaardepapierenZaakAction",
   "$id": "https://waardepapieren.commonground.nl/action/waar.WaardepapierenZaakAction.action.json",
   "$schema": "https://docs.commongateway.nl/schemas/Action.schema.json",
-  "version": "0.0.2",
-  "listens": [
-    "commongateway.object.create",
-    "commongateway.object.update"
-  ],
+  "version": "0.0.10",
```
Comment (bbrands02) [inline Installation/Action/waar.WaardepapierenZaakAction.action.json]: 0.0.3?

File: Installation/Action/waar.WaardepapierenZaakAction.action.json [hunk 3]
```diff
@@ -2,39 +2,30 @@
...
       },
-      {
-        "==": [
-          {
-            "var": "entity"
-          },
-          "https://vng.opencatalogi.nl/schemas/zrc.rol.schema.json"
-        ]
-      }
+      "https://open-zaak.test.buren.opengem.nl/catalogi/api/v1/zaaktypen/f0e9be33-10ae-43f1-bc1d-9f1bb4579fd4"
```
Comment (bbrands02) [inline Installation/Action/waar.WaardepapierenZaakAction.action.json:14]: Should this be configured or is it ok to have a url here?

File: Installation/Mapping/drc.enkelvoudigInformatieObjectUpstream.mapping.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "title": "EnkelvoudigInformatieobject",
+  "$id": "https://waardepapieren.commongateway.nl/mapping/drc.enkelvoudigInformatieObjectUpstream.mapping.json",
+  "$schema": "https://docs.commongateway.nl/schemas/Mapping.schema.json",
+  "version": "0.0.2",
```
Comment (bbrands02) [inline Installation/Mapping/drc.enkelvoudigInformatieObjectUpstream.mapping.json]: 0.0.1?

File: Installation/Mapping/zrc.zaakInformatieObjectUpstream.mapping.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "title": "Zaakinformatieobject",
+  "$id": "https://waardepapieren.commongateway.nl/mapping/zrc.zaakInformatieObjectUpstream.mapping.json",
+  "$schema": "https://docs.commongateway.nl/schemas/Mapping.schema.json",
+  "version": "0.0.2",
```
Comment (bbrands02) [inline Installation/Mapping/zrc.zaakInformatieObjectUpstream.mapping.json]: 0.0.1?

File: src/Service/WPZaakService.php [hunk 6]
```diff
@@ -137,86 +217,392 @@ public function saveWaardepapierInDRC(string $data, ObjectEntity $zaakObject): v
...
+        }
+
+        // Check if we have a zaaktype with findSyncByObject.
+        $zaaktypeSync = $this->syncService->findSyncBySource($source, $schema, $zaaktypeId);
+
+        // if no object is present, a call must be made to retrieve the zaaktype.
+        try {
+            $response = $this->callService->call($source, '/zaaktypen/'.$zaaktypeId);
+        } catch (\Exception $exception) {
+            // Throw error.
```
Comment (bbrands02) [inline src/Service/WPZaakService.php]: Note: no error is thrown

File: src/Service/WPZaakService.php [hunk 7]
```diff
@@ -137,86 +217,392 @@ public function saveWaardepapierInDRC(string $data, ObjectEntity $zaakObject): v
...
+
+        // Check if we have a zaaktype with findSyncByObject.
+        $objectSync = $this->syncService->findSyncBySource($source, $schema, $objectId);
+
+        // if no object is present, a call must be made to retrieve the zaaktype.
+        if ($objectSync->getObject() === null) {
+            try {
+                $response = $this->callService->call($source, $endpoint.'/'.$objectId);
+            } catch (\Exception $exception) {
+                // Throw error.
```
Comment (bbrands02) [inline src/Service/WPZaakService.php]: Note: no error is thrown

File: src/Service/WPZaakService.php [hunk 8]
```diff
@@ -137,86 +217,392 @@ public function saveWaardepapierInDRC(string $data, ObjectEntity $zaakObject): v
...
+     * @return ObjectEntity|null The zaaktype of the given source
+     */
+    public function getZaakFromSource(ObjectEntity $zaak): ?ObjectEntity
+    {
+        $source = $this->resourceService->getSource($this->configuration['zrcSource'], 'common-gateway/waardepapieren-bundle');
+
+        try {
+            $response = $this->callService->call($source, '/zaken');
+        } catch (\Exception $exception) {
+            // Throw error.
```
Comment (bbrands02) [inline src/Service/WPZaakService.php]: Note: no error is thrown

**Review by bbrands02 — APPROVED** [review]

**Pull request merged**